packages/beacon-node/src/network/network.ts

```diff
@@ -350,4 +358,34 @@ export class Network implements INetwork {
       }
     }
   };
+
+  private onLightClientFinalityUpdate = async (finalityUpdate: altair.LightClientFinalityUpdate): Promise<void> => {
+    try {
+      // messages SHOULD be broadcast after one-third of slot has transpired
+      // https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#sync-committee
+      await this.waitOneThirdOfSlot(finalityUpdate.signatureSlot);
+      return await this.gossip.publishLightClientFinalityUpdate(finalityUpdate);
+    } catch (e) {
+      this.logger.debug("Error on BeaconGossipHandler.onLightclientFinalityUpdate", {}, e as Error);
+    }
+  };
+
+  private onLightClientOptimisticUpdate = async (
+    optimisticUpdate: altair.LightClientOptimisticUpdate
+  ): Promise<void> => {
+    try {
+      // messages SHOULD be broadcast after one-third of slot has transpired
+      // https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#sync-committee
+      await this.waitOneThirdOfSlot(optimisticUpdate.signatureSlot);
+      return await this.gossip.publishLightClientOptimisticUpdate(optimisticUpdate);
+    } catch (e) {
+      this.logger.debug("Error on BeaconGossipHandler.onLightclientOptimisticUpdate", {}, e as Error);
+    }
+  };
+
+  private waitOneThirdOfSlot = async (slot: number): Promise<void> => {
+    const minPubTime = computeTimeAtSlot(this.config, slot, this.chain.genesisTime) + this.config.SECONDS_PER_SLOT / 3;
+    const waitTime = minPubTime - Date.now() / 1000;
+    await sleep(waitTime);
```

Comment:
I think sleep takes a second argument, an abort signal. Can you add the signal from the constructor here?
Without that, if we're sleeping here and we exit lodestar, we may still have to sleep the full duration.